src/models/version.rs

```diff
@@ -212,8 +212,18 @@ fn validate_license_expr(s: &str) -> AppResult<()> {
         allow_postfix_plus_on_gpl: true,
     };
 
-    spdx::Expression::parse_mode(s, PARSE_MODE).map_err(|_| {
-        cargo_err("unknown or invalid license expression; see http://opensource.org/licenses for options, and http://spdx.org/licenses/ for their identifiers")
+    spdx::Expression::parse_mode(s, PARSE_MODE).map_err(|e| {
+        cargo_err(&format_args!(
+            "unknown or invalid license expression; \
+            see http://opensource.org/licenses for options, \
+            and http://spdx.org/licenses/ for their identifiers\n\
+            Note: If you have a non-standard license that is not listed by SPDX, \
+            use the license-file field to specify the path to a file containing \
+            the text of the license.\n\
+            See https://doc.rust-lang.org/cargo/reference/manifest.html#the-license-and-license-file-fields \
+            for more information.\n\
+            {e}"
```

Comment:
since we are including the original SPDX error now, could you add a test case for this? :)